### ISSUE TITLE
It would be good to allow suppression of repositories.

### ISSUE THREAD

**Issue by tkruse** [ben-manes/gradle-versions-plugin#30]
Hi,my problem is that when using the version plugin and the grunt plugin like this:

    apply plugin: 'versions'
    apply plugin: 'grunt' // and grunt_tasks. implies node plugin
    buildscript {
    repositories {
        jcenter() // bintray jcenter faster than mavenCentral
        maven { url 'http://dl.bintray.com/robfletcher/gradle-plugins' }
    }
    dependencies {
        classpath 'com.moowork.gradle:gradle-grunt-plugin:0.5'
    }
    }
    node {
      version = '0.10.20'
      download = true
      workDir = file("${projectDir}/nodejs") // Set the work directory for unpacking node
    }

then the dependencyUpdates Task takes an eternity because the each dependency is tried 100 times against the nodejs repo:

     $ gradle dependencyUpdates --info
      Task has not declared any outputs.
    Resolving with repositories:
     - BintrayJCenter: http://jcenter.bintray.com/
    ...
     - ivy: http://nodejs.org/dist
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.13/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.13/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.12/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.12/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.11/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.11/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.10/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.10/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.9/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.9/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.8/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.8/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.7/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.7/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.6/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.6/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.5/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.5/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.4/ivy.xml]
    Resource missing. [HTTP HEAD: http://nodejs.org/dist/v0.11.4/powermock-classloading-xstream.jar]
    Resource missing. [HTTP GET: http://nodejs.org/dist/v0.11.3/ivy.xml]
    ...

Not sure how this can best be prevented. The repo is actually added in the gradle_node_plugin:
https://github.com/srs/gradle-node-plugin/blob/master/src/main/groovy/com/moowork/gradle/node/NodePlugin.groovy

**Comment by ben-manes** [comment]
We have two options and I am happy with either. Which do you prefer?

1. Aggregate and resolve with all repositories
The current strategy was used due to resolving all dependencies together in a detached configuration. This has the widest detection field.

2. Use repositories associated with a configuration
This is Gradle's normal behavior and isolates a problematic repository to the trusted usages. The plugin now uses a aggregates the results across all configurations, so all that is needed is to remove the repository manipulation logic.

In v0.12 I retained the behavior of aggregating repositories for backwards compatibility. If anyone prefers that we drop this then that should be an easy change.

**Comment by jochenberger** [comment]
I know, I'm probably late for the discussion, but I often find myself wishing there was a way to restrict the latest version lookup on a per-dependency basis.
I don't mean to hijack this issue, but I think this is related.
At my company, we have a lot of internal libraries and I know that those are only available from a certain repository and there is no point in looking them up in Maven Central. So I'd like to be able to tell the plugin (or Gradle itself?) to look those up in our internal repository only.

**Comment by ben-manes** [comment]
@jochenberger I think this is best done by only using the local repository in Gradle, which acts as a proxy to the remote repositories. Artifactory's virtual repository supports [include/exclude patterns](https://www.jfrog.com/confluence/display/RTF/Virtual+Repositories#VirtualRepositories-UsingIncludesandExcludesPatterns) and uses this optimization as the primary example.

**Comment by tkruse** [comment]
I do not understand the strategies, but nor can I reproduce the problem with current versions of the plugins.

What I get now is even weirder:
```
apply plugin: 'com.github.ben-manes.versions'
apply plugin: 'com.moowork.grunt'

// In this section you declare where to find the dependencies of your project
repositories {
    // Use 'jcenter' for resolving your dependencies.
    // You can declare any Maven/Ivy/file repository here.
    jcenter()
}

buildscript {
    repositories {
        jcenter()
        maven { url 'http://dl.bintray.com/robfletcher/gradle-plugins' }
    }

    dependencies {
        classpath 'com.github.ben-manes:gradle-versions-plugin:0.11.3'
        classpath 'com.moowork.gradle:gradle-grunt-plugin:0.10'
    }
}

node {
    version = '0.10.20'
    download = true
    workDir = file("${projectDir}/nodejs") // Set the work directory for unpacking node
}
```

Result:
```
$ gradle dependencyUpdates  --info
...
Failed to determine the latest version for the following dependencies (use --info for details):
 - org.nodejs:node
The exception that is the cause of unresolved state: Could not find any matches for org.nodejs:node:+ as no versions of org.nodejs:node are available.
Searched in the following locations:
    https://jcenter.bintray.com/org/nodejs/node/maven-metadata.xml
    https://jcenter.bintray.com/org/nodejs/node/
    http://dl.bintray.com/robfletcher/gradle-plugins/org/nodejs/node/maven-metadata.xml
    http://dl.bintray.com/robfletcher/gradle-plugins/org/nodejs/node/
    http://nodejs.org/dist/
    http://nodejs.org/dist/v0.12.5/ivy.xml
    http://nodejs.org/dist/v0.12.5/node.jar
```

But when I change the node version to 0.12.x, I get:
```
...
Selection of org.nodejs:node:0.12.5 rejected by component selection rule: Component status integration rejected by revision milestone
Selection of org.nodejs:node:0.12.4 rejected by component selection rule: Component status integration rejected by revision milestone
Selection of org.nodejs:node:0.12.3 rejected by component selection rule: Component status integration rejected by revision milestone
Selection of org.nodejs:node:0.12.2 rejected by component selection rule: Component status integration rejected by revision milestone
Selection of org.nodejs:node:0.12.1 rejected by component selection rule: Component status integration rejected by revision milestone
Selection of org.nodejs:node:0.12.0 rejected by component selection rule: Component status integration rejected by revision milestone
...
Failed to determine the latest version for the following dependencies (use --info for details):
 - org.nodejs:node
The exception that is the cause of unresolved state: Could not find any version that matches org.nodejs:node:+.
Versions rejected by component selection rules:
    0.12.5
    0.12.4
    0.12.3
    0.12.2
    0.12.1
Searched in the following locations:
    https://jcenter.bintray.com/org/nodejs/node/maven-metadata.xml
    https://jcenter.bintray.com/org/nodejs/node/
...
```

I have no idea what is going on there...

**Comment by ben-manes** [comment]
#66 means that we'll probably have to stop aggregating all repositories, which will remove the original problem.

It looks like the fake ivy repository is defaulting all versions to the `integration` status, so they are being rejected at the `milestone` status constraint. I'll have to look at it and see if that's true, but sounds like a side effect of the node repository hack.

**Comment by ben-manes** [comment]
The plugin author has fixed the issue by dynamically adding the npm repository. In addition, the fix for #66 stops aggregating repositories so this shouldn't be a problem anymore
